src/sentry/issues/issue_velocity.py

```diff
@@ -0,0 +1,203 @@
+"""This module contains the logic for calculating the velocity threshold based on the 90th percentile
+of events per issue per hour, and getting it from and saving it to Redis.
+"""
+
+from __future__ import annotations
+
+import logging
+import math
+from datetime import datetime, timedelta
+from typing import Optional
+
+from django.conf import settings
+from sentry_redis_tools.clients import RedisCluster, StrictRedis
+from snuba_sdk import (
+    Column,
+    Condition,
+    Direction,
+    Entity,
+    Function,
+    Limit,
+    Op,
+    OrderBy,
+    Query,
+    Request,
+)
+
+from sentry.locks import locks
+from sentry.models.project import Project
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.utils import json
+from sentry.utils.locking import UnableToAcquireLock
+from sentry.utils.redis import redis_clusters
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+
+# for snuba operations
+REFERRER = "sentry.issues.issue_velocity"
+THRESHOLD_QUANTILE = {"name": "p90", "function": "quantile(0.9)"}
+WEEK_IN_HOURS = 7 * 24
+
+# for redis operations
+REDIS_TTL = 24 * 60 * 60  # 1 day
```

Comment:
Since we're updating once a day, it might make sense for this ttl to be > 1 day so that we can read the stale values for a while.

Reply:
gotcha, i guess is there any particular number we had in mind?

Reply:
Maybe 2 days? The exact number probably isn't too important